File: PythonDataAnalysis/Merging Data.R
# Read in data
Part_dat=read.csv("Desktop/PythonLearning/Pythonlearning/PythonDataAnalysis/Dataset_Partial.csv")
Bamt_dat=read.csv("Desktop/PythonLearning/Pythonlearning/PythonDataAnalysis/Dataset_BenAmt.csv")
Nfam_dat=read.csv("Desktop/PythonLearning/Pythonlearning/PythonDataAnalysis/Dataset_NoFamily.csv")

# Check dimensions
dim(Part_dat)
dim(Bamt_dat)
dim(Nfam_dat)

# See column names
colnames(Part_dat)
colnames(Bamt_dat)
colnames(Nfam_dat)

# View first few observations
head(Part_dat,n = 4)

dat_temp = merge(Part_dat,Bamt_dat,by="ID",all.x=T)
dat_temp1 = merge(Part_dat,Bamt_dat,by=c("ID","Payment_Date"),all.x=T)
dat_temp2 = merge(Part_dat, Nfam_dat, by.x="ID",by.y="KEY",all.x=T)
dat_temp3 = merge(dat_temp1,Nfam_dat,by.x="ID",by.y="KEY",all.x=T)

write.csv(dat_temp3,"Desktop/PythonLearning/Pythonlearning/PythonDataAnalysis/R_Merged_Data.csv")

File: PythonDataAnalysis/Merging Data.py
import pandas as pd
import os

# read in data
Part_dat = pd.read_csv('Dataset_Partial.csv',header=0)
Bamt_dat = pd.read_csv('Dataset_BenAmt.csv',header=0)
Nfam_dat = pd.read_csv('Dataset_NoFamily.csv',header=0)

# get dimensions
Part_dat.shape
Bamt_dat.shape
Nfam_dat.shape

# get column names
Part_dat.columns
Bamt_dat.shape
Nfam_dat.shape

# get first few observations
Part_dat.head(4)

# Database Style Merging
dat_temp = pd.merge(Part_dat,Bamt_dat,how='left', on = ['ID']) # Payment_Date didn't merge together
dat_temp.shape
# duplicates, to avoide, merge based on multiple columns
dat_temp1 = pd.merge(Part_dat, Bamt_dat, how='left', on=['ID','Payment_Date'])
dat_temp1.shape

# failed: data_temp3 = pd.merge(Part_dat, Nfam_dat, how= 'left', on=['ID'])
data_temp2 = pd.merge(Part_dat, Nfam_dat, how= 'left', left_on=['ID'], right_on=['KEY'])
dat_temp3 = pd.merge(dat_temp1, Nfam_dat, how='left', left_on=['ID'], right_on=['KEY'])
dat_temp3.head()

# export data
dat_temp3.to_csv("Merged_Data.csv",sep=",",header=True,index=False)